src/v2/view-builder/views/ov/ResendNumberChallengeView.js

```diff
@@ -0,0 +1,16 @@
+import { View } from 'okta';
+import hbs from 'handlebars-inline-precompile';
+
+const ResendNumberChallengeView = View.extend({
```

Comment:
I was wondering better would be to let this view show/clear warning instead of `NumberChallengePushView.js`.. main view can trigger events to do that.. Basically `this.warningTimeout` can completely resides in this view.

Reply:
Yeah, I thought about that but it feels like it adds more complexity and I was trying to avoid that. To do this it would mean that the `ResendNumberChallenge` view would also need to know when polling started and stopped, for which I would have to trigger a few more events and have listeners in the `ResendNumberChallengeView`.

Reply:
`ResendNumberChallenge` should not know about polling.. it can just receive an event when to start and stop right?

```
    startPoll () {
      this.startPolling();
      this.trigger('toggleWarning'); //can handle start and stop 
    }
```

Reply:
@pradeepdewda-okta  There are a few things that will need to happen for this 

-  The `ResendNumberChallengeView ` will be added immediately, gets hidden on load and is then listening to 
`showWarning`
-  `NumberChallengePushView`  then triggers `showWarning` 
-  `ResendNumberChallengeView` will then need to start the timeout and shows the warning. 
-  `NumberChallengePushView` will trigger another evert to hide the warning when user hits resend.

These will all be events on the `appState` as `state` is not available within these views. Let me know if this is inline with what you are thinking.